src/components/Form/InlineEditFormControl.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop } from '../../common/helpers';
+
+import { InlineEdit, ConfirmButton, CancelButton } from '../InlineEdit';
+import { Spinner } from '../Spinner';
+import { Form, FormControl, InputGroup } from './index';
+
+class InlineEditFormControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: props.editing,
+      changed: props.changed,
+      value: props.value
+    };
+
+    this.inlineEditController = {
+      isEditing: () => this.state.editing,
+      onActivate: e => {
+        this.setState({
+          editing: true,
+          backup: this.state.value
+        });
+      },
+      onConfirm: () => {
+        this.setState({
+          editing: false,
+          changed: false
+        });
+        this.props.onConfirm(this.state.value);
+      },
+      onCancel: () => {
+        this.setState({
+          editing: false,
+          changed: false,
+          value: this.state.backup,
+          backup: null
+        });
+        this.props.onCancel();
+      },
+      onChange: value => {
+        this.setState({
+          changed: true,
+          value
+        });
+      }
+    };
+  }
+
+  renderValue(value) {
+    return (
+      <InputGroup className="pf-inline-edit">
+        <span
+          onClick={e => this.inlineEditController.onActivate(e)}
+          className="value edit-icon-permanent"
+        >
+          {value}
+        </span>
+      </InputGroup>
+    );
+  }
+
+  renderEdit(editValue) {
+    const {
+      value,
+      changed,
+      updating,
+      editing,
+      onConfirm,
+      ...controlProps
+    } = this.props;
+    let buttons;
+    if (updating) {
+      buttons = <Spinner loading inline size="xs" />;
+    } else {
+      buttons = [
+        <ConfirmButton
+          bsStyle="primary"
```

Comment:
The `primary` styling should be on the ConfirmButton so that any use outside of the `InlineEditFormControl` gets the proper styling.